tuf/client/updater.py

```diff
@@ -1493,47 +1580,58 @@ def _update_metadata_if_changed(self, metadata_role,
     # untrusted data is not decompressed prior to verifying hashes, or
     # decompressing a file that may be invalid or partially intact.
     compression = None
-    compressed_fileinfo = None
 
-    # Extract the fileinfo of the uncompressed version of 'metadata_role'.
-    uncompressed_fileinfo = self.metadata['current'][referenced_metadata] \
-                                         ['meta'] \
-                                         [uncompressed_metadata_filename]
+    # Extract the versioninfo of the uncompressed version of 'metadata_role'.
+    expected_versioninfo = self.metadata['current'][referenced_metadata] \
+                                        ['meta'] \
+                                        [uncompressed_metadata_filename]
 
     # Check for the availability of compressed versions of 'snapshot.json',
     # 'targets.json', and delegated Targets (that also start with 'targets').
     # For 'targets.json' and delegated metadata, 'referenced_metata'
     # should always be 'snapshot'.  'snapshot.json' specifies all roles
-    # provided by a repository, including their file lengths and hashes.
+    # provided by a repository, including their version numbers.
     if metadata_role == 'snapshot' or metadata_role.startswith('targets'):
```

Comment:
(Caveat: this is not related to this change and doesn't need to be fixed for this PR, but I think it should be fixed at some point, if I understand correctly.)

I dislike the whole split-conditional structure here.
(If conditional 1: make temp variable for use if conditionals 1 and 2. If conditional 2: continue logic.)

If I understand correctly, this is better:

```
    if ( 'gzip' in self.metadata['current']['root']['compression_algorithms']
        and (metadata_role == 'snapshot' or metadata_role.startswith('targets'))
        ):
      gzip_metadata_filename = uncompressed_metadata_filename + '.gz'
      compression = 'gzip'
      logger.debug('Compressed version of ' +
                repr(uncompressed_metadata_filename) + ' is available at ' +
                repr(gzip_metadata_filename) + '.')
    else:
        logger.debug('Compressed version of ' +
                     repr(uncompressed_metadata_filename) + ' not available.')
```

* There is no longer a case where the metadata role is snapshot or starts with targets but gzip is not an accepted compresison algorithm in which the temp gzip_metadata_filename is created. (Is this actually needed in some way? If so, I'd say suspect that might be worse.)
* Removed the "\"s in favor of parenthetical-based line continuation.
* Unfortunately had to leave a "):" on its own line because of line lengths.

It's correct syntax in the frame I've stuck it in, but I haven't tried to run tests on it. I'll loop back to try it out another time so I'm sure it doesn't break anything. LMK if I'm wrong at your leisure. (: